### PR TITLE
Typed rib

### PR DESCRIPTION
**Pull request by afsalthaj**

### CODE REVIEW

File: golem-rib/src/type_inference/expr_visitor.rs [hunk 1]
```diff
@@ -0,0 +1,206 @@
+use crate::Expr;
+use std::collections::VecDeque;
+use std::ops::Deref;
+
```
Comment (afsalthaj) [inline golem-rib/src/type_inference/expr_visitor.rs:4]: I could refactor this, but somewhere Rust gave me hard time

File: golem-api-grpc/proto/golem/apidefinition/api_definition.proto [hunk 2]
```diff
@@ -45,8 +58,21 @@ enum HttpMethod {
 }
 
 message WorkerBinding {
-  golem.component.ComponentId component = 1;
+  golem.component.VersionedComponentId component = 1;
```
Comment (justcoon) [inline golem-api-grpc/proto/golem/apidefinition/api_definition.proto:61]: just small detail, we should rename component -> component_id  (to match model naming)

File: golem-api-grpc/proto/golem/apidefinition/api_definition.proto [hunk 3]
```diff
@@ -45,8 +58,21 @@ enum HttpMethod {
...
   golem.rib.Expr worker_name = 2;
   golem.rib.Expr response = 3;
   optional golem.rib.Expr idempotency_key = 4;
-}
\ No newline at end of file
+}
+
+message CompiledWorkerBinding {
+  golem.component.VersionedComponentId component = 1;
```
Comment (justcoon) [inline golem-api-grpc/proto/golem/apidefinition/api_definition.proto:68]: just small detail, we should rename component -> component_id

**Review by vigoo — APPROVED** [review]

File: golem-worker-service-base/src/service/http/http_api_definition_validator.rs [hunk 4]
```diff
@@ -15,7 +14,7 @@ use crate::service::api_definition_validator::{ApiDefinitionValidatorService, Va
 pub struct RouteValidationError {
     pub method: MethodPattern,
     pub path: String,
-    pub component: ComponentId,
+    pub component: VersionedComponentId,
```
Comment (justcoon) [inline golem-worker-service-base/src/service/http/http_api_definition_validator.rs:17]: this should be also renamed  
component -> component_id

**Review by justcoon — APPROVED** [review]

**Pull request merged**